subsys/bluetooth/audio/tbs_client.c

```diff
@@ -0,0 +1,2166 @@
+	struct bt_gatt_subscribe_params friendly_name_sub_params;
+	struct bt_gatt_discover_params friendly_name_sub_disc_params;
+
+	struct bt_gatt_read_params read_params;
+	uint8_t read_buf[BT_ATT_MAX_ATTRIBUTE_LEN];
+	struct net_buf_simple net_buf;
+};
+
+struct bt_tbs_server_inst {
+	struct bt_tbs_instance tbs_insts[BT_TBS_INSTANCE_MAX_CNT];
+	struct bt_gatt_discover_params discover_params;
+	struct bt_tbs_instance *current_inst;
+	uint8_t inst_cnt;
+	bool gtbs_found;
+	bool subscribe_all;
+};
+
+static struct bt_tbs_client_cb *tbs_client_cbs;
+
+static struct bt_tbs_server_inst srv_insts[CONFIG_BT_MAX_CONN];
+static const struct bt_uuid *tbs_uuid = BT_UUID_TBS;
+static const struct bt_uuid *gtbs_uuid = BT_UUID_GTBS;
+
+static void discover_next_instance(struct bt_conn *conn, uint8_t index);
+
+static bool valid_inst_index(struct bt_conn *conn, uint8_t idx)
+{
+	uint8_t conn_index;
+
+	__ASSERT(conn, "NULL conn");
+
+	conn_index = bt_conn_index(conn);
+
+	if (IS_ENABLED(CONFIG_BT_TBS_CLIENT_GTBS) && idx == BT_TBS_GTBS_INDEX) {
+		return true;
+	} else {
+		return idx < srv_insts[conn_index].inst_cnt;
+	}
+}
+
+static struct bt_tbs_instance *get_inst_by_index(struct bt_conn *conn,
+						 uint8_t idx)
+{
+	uint8_t conn_index;
+
+	__ASSERT(conn, "NULL conn");
+
+	conn_index = bt_conn_index(conn);
+
+	if (IS_ENABLED(CONFIG_BT_TBS_CLIENT_GTBS) && idx == BT_TBS_GTBS_INDEX) {
+		return &srv_insts[conn_index].tbs_insts[GTBS_INDEX];
+	} else {
+		return &srv_insts[conn_index].tbs_insts[idx];
+	}
+}
+
+static bool free_call_spot(struct bt_tbs_instance *inst)
+{
+	for (int i = 0; i < CONFIG_BT_TBS_CLIENT_MAX_CALLS; i++) {
+		if (inst->calls[i].index == BT_TBS_FREE_CALL_INDEX) {
+			return true;
+		}
+	}
+	return false;
+}
+
+static struct bt_tbs_instance *lookup_inst_by_handle(struct bt_conn *conn,
+							 uint16_t handle)
+{
+	uint8_t conn_index;
+	struct bt_tbs_server_inst *srv_inst;
+
+	__ASSERT(conn, "NULL conn");
+
+	conn_index = bt_conn_index(conn);
+	srv_inst = &srv_insts[conn_index];
+
+	for (size_t i = 0; i < ARRAY_SIZE(srv_inst->tbs_insts); i++) {
+		if (srv_inst->tbs_insts[i].start_handle <= handle &&
+		    srv_inst->tbs_insts[i].end_handle >= handle) {
+			return &srv_inst->tbs_insts[i];
+		}
+	}
+	BT_DBG("Could not find instance with handle 0x%04x", handle);
+
+	return NULL;
+}
+
+static uint8_t net_buf_pull_call_state(struct net_buf_simple *buf,
+				       struct bt_tbs_client_call_state *call_state)
+{
+	if (buf->len < sizeof(*call_state)) {
+		BT_DBG("Invalid buffer length %u", buf->len);
+		return BT_ATT_ERR_INVALID_ATTRIBUTE_LEN;
+	}
+
+	call_state->index = net_buf_simple_pull_u8(buf);
+	call_state->state = net_buf_simple_pull_u8(buf);
+	call_state->flags = net_buf_simple_pull_u8(buf);
+
+	return 0;
+}
+
+
+static uint8_t net_buf_pull_call(struct net_buf_simple *buf,
+				 struct bt_tbs_client_call *call)
+{
+	uint8_t item_len;
+	uint8_t uri_len;
+	const size_t mimimum_item_len = sizeof(call->call_info) + BT_TBS_MIN_URI_LEN;
```

Comment:
Seems like a misspelling of "minimum". Also, I'd just call it `min_item_len`. Also, move it as the first variable definition instead of having a long line in the middle of the definitions.